spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java

```diff
@@ -448,6 +448,12 @@ public void testInvalidTimeTravelAgainstBranchIdentifierWithAsOf() {
             () -> sql("SELECT * FROM %s.branch_b1 VERSION AS OF %s", tableName, snapshotId))
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("Cannot do time-travel based on both table identifier and AS OF");
+
+    // using branch_b1 in the table identifier and TIMESTAMP AS OF
+    Assertions.assertThatThrownBy(
+            () -> sql("SELECT * FROM %s.branch_b1 TIMESTAMP AS OF %s", tableName, snapshotId))
```

Comment:
[minor] should we pass actual timestamp instead of snapshotId ? though I agree this error is hit before the timestamp parsing so testing like this is also sufficient.

Reply:
Good catch @singhpk234 I totally missed this. Would be good to fix this test to pass a timestamp just so someone reading it isn't confused!

Reply:
oops that was a copy/paste error. I've changed this to using `now()`